lib/xcode/install.rb

```diff
@@ -6,35 +6,108 @@
 require 'rubygems/version'
 require 'xcode/install/command'
 require 'xcode/install/version'
+require 'shellwords'
+require 'open3'
+require 'fileutils'
 
 module XcodeInstall
   CACHE_DIR = Pathname.new("#{ENV['HOME']}/Library/Caches/XcodeInstall")
   class Curl
     COOKIES_PATH = Pathname.new('/tmp/curl-cookies.txt')
 
-    def fetch(url, directory = nil, cookies = nil, output = nil, progress = true)
+    # @param url: The URL to download
+    # @param directory: The directory to download this file into
+    # @param cookies: Any cookies we should use for the download (used for auth with Apple)
+    # @param output: A PathName for where we want to store the file
+    # @param progress: parse and show the progress?
+    # @param progress_block: A block that's called whenever we have an updated progress %
+    #                        the parameter is a single number that's literally percent (e.g. 1, 50, 80 or 100)
+    # rubocop:disable Metrics/AbcSize
+    def fetch(url: nil,
+              directory: nil,
+              cookies: nil,
+              output: nil,
+              progress: nil,
+              progress_block: nil)
+
       options = cookies.nil? ? [] : ['--cookie', cookies, '--cookie-jar', COOKIES_PATH]
-      # options << ' -vvv'
 
       uri = URI.parse(url)
       output ||= File.basename(uri.path)
       output = (Pathname.new(directory) + Pathname.new(output)) if directory
 
       retry_options = ['--retry', '3']
-      progress = progress ? '--progress-bar' : '--silent'
-      command = ['curl', *options, *retry_options, '--location', '--continue-at', '-', progress, '--output', output, url].map(&:to_s)
+      command = [
+        'curl',
+        *options,
+        *retry_options,
+        '--location',
+        '--continue-at',
+        '-',
+        '--output',
+        output,
+        url
+      ].map(&:to_s)
+
+      progress_log_file = File.join(CACHE_DIR, "progress.#{Time.now.to_i}.progress")
+      FileUtils.rm(progress_log_file) if File.exist?(progress_log_file)
```

Comment:
this could be `rm_f` instead of checking if it exists

Reply:
Nice, will update